lib/account.rb

```diff
@@ -0,0 +1,83 @@
+require 'csv'
+require 'time'
+
+module Bank
+  class Account
+    # attr_accessor :withdraw, :deposit, :all_accounts
+    attr_reader :id, :balance, :opendate, :show_id, :account_count, :find_account
+    @@count = 0
+
+    def initialize(id, balance, opendate)
+      @@count += 1
+      @id = id.to_i
+      @balance = monify(balance)
+      unless @balance >= 0
+        raise ArgumentError.new "Starting balance is not valid."
+      end
+      @opendate = Time.parse(opendate)
+      @withdraw_fee = 0
+      @minimum_balance = 0
+      # @opendate = DateTime.strptime(opendate, '%Y-%m-%d %H:%M:%S %z')
+    end
+
+    # check if balance is in cents or in dollars with '.'
+    def monify(amount)
+      float_already = false
+      amount.to_s.each_char do |letter|
+        if letter == "."
+          float_already = true
+        else
+        end
+      end
+      if float_already == false
+        return amount.to_i/100.0
+      else
+        return amount.to_f
+      end
+    end
+
+    def withdraw(amount)
+      unless amount > 0
+        raise ArgumentError.new "Withdrawal has be a positive amount!"
+      end
+
+      if @balance - (amount + @withdraw_fee) < @minimum_balance
+        puts "Can't withdraw more than you have!"
+      else
+        @balance -= (amount + @withdraw_fee)
+      end
+      return @balance
+    end
+
+    def deposit(amount)
+      unless amount > 0
+        raise ArgumentError.new "Must deposit a positive amount!"
+      end
+      @balance += amount
+      return @balance
+    end
+
+    def self.all
+      @all_accounts = []
```

Comment:
I'm curious why you chose to use an instance variable here. It seems to me a local variable is the right choice, since you don't ever reference the saved value.

It's also worth noting that using an instance variable inside a class method does something subtly different than what you might expect. Best to stick with class variables (two @s) in class methods, and instance variables (one @) in instance methods.